### ISSUE TITLE
Upgrading for Unity 5.0

### ISSUE THREAD

**Issue by iant-ee** [unitycontainer/documentation#5]
I am upgrading from Unity 3.5. As a first step I'm moving to Unity 5.0, but the documentation here only starts from 5.2.1.

I'm going to add notes here of any problems I encounter.

**Comment by iant-ee** [comment]
I had to upgrade the Unity.Mvc package to 5.0 and replace
```
using Microsoft.Practices.ObjectBuilder2;
using Microsoft.Practices.Unity;
Microsoft.Practices.Unity.Configuration;
Microsoft.Practices.Unity.Mvc;
```
with
```
using Unity;
using Unity.Attributes;
using Unity.Interception.Utilities;
using Unity.Mvc
```

**Comment by iant-ee** [comment]
We were using Lydian.Unity.Automapper. This project hasn't been updated in many years and doesn't looks to be compatible with Unity 5.

https://blogs.msdn.microsoft.com/agile/2013/03/12/unity-configuration-registration-by-convention/ suggests that it is no longer needed, as Unity itself now supports automapping.

**Comment by ENikS** [comment]
I would suggest starting with at least 5.8.6. Anything prior to that version was quite unstable. 
… thinking about it, why not go for the latest?

**Comment by iant-ee** [comment]
I'm not planning on going live on an old version, but when I upgrade my project won't compile. By upgrading in small increments I'm hoping the problems I encounter will be limited to the changes in that release. As soon as my code compiles and I can click around a few pages of the app I was intending to move to a newer version.

To start with, I was going to use any version less that 5.2.1, so I don't need to make the changes described at https://github.com/unitycontainer/unity/wiki

Does that make sense, or do you still think I should go straight to the latest?

**Comment by ENikS** [comment]
This path has been [documented](https://unitycontainer.github.io/articles/changelog.html) and lists most of the issues you are going to deal with.

Instead of repeating [this issue](https://github.com/unitycontainer/unity/issues/159) perhaps you could improve posted wiki with your findings?

**Comment by iant-ee** [comment]
Thanks. I guess my path will be 5.2.0 > 5.2.1 > 5.8.0 > 5.9.0 > latest.

**Comment by ENikS** [comment]
Closing this as a duplicate of [#159](https://github.com/unitycontainer/unity/issues/159)

**Comment by iant-ee** [comment]
I'll continue using this issue to add my notes, but I'll try and submit a pull request for the changelog if I come up with clear conclusions.

I have been able to successfully upgrade to 5.3.1. Moving to 5.3.2 I get "Method 'GetValue' in type 'Unity.Mvc.PerRequestLifetimeManager' from assembly 'Unity.Mvc, Version=5.0.0.0, Culture=neutral, PublicKeyToken=6d32ff45e0ccc69f' does not have an implementation."

5.3.1 doesn't have a tag, but changes from 5.3.0 to 5.3.2 are https://github.com/unitycontainer/unity/compare/5.3.0.313...5.3.2.353

Reading https://github.com/unitycontainer/unity/issues/124 I think that although my app was compiling and running, PerRequestLifetimeManager wasn't actually doing what it should on earlier 5.x releases.

**Comment by iant-ee** [comment]
PerRequestLifetimeManager has moved to the Unity.Mvc package. I had 5.0.0 installed, by upgrading that to 5.0.10 (released around the same time as Unity 5.3.2) my app is working again.
